sdks/java/expansion-service/src/main/java/org/apache/beam/sdk/expansion/service/ExpansionService.java

```diff
@@ -317,6 +321,14 @@ default InputT createInput(Pipeline p, Map<String, PCollection<?>> inputs) {
     Pipeline pipeline = Pipeline.create();
     ExperimentalOptions.addExperiment(
         pipeline.getOptions().as(ExperimentalOptions.class), "beam_fn_api");
+    List<String> classpathResources =
+        detectClassPathResourcesToStage(Environments.class.getClassLoader(), pipeline.getOptions());
```

Comment:
This assumes that the set of classpath resources required to execute Java SDK harness and execute the cross-langauge Java step to be the same as the set of classpath resources specified when starting up the expansion service, right ?

I'm wondering if we can always rely on this. At least we should document this so that users are aware how to stage additional resources if needed.

Reply:
Correct. This is a reasonable default (and the provider of an expansion service can always arrange that it is so), but we'll probably eventually want a way to customize this.